tools/templates/DeviceBindingTemplate/dotnet_new_device-binding_csharp/.template.config/template.json

```diff
@@ -0,0 +1,14 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "IoT Device Binding",
+  "classifications": [ "IoT", "Console", "Library", "Test" ],
+  "identity": "DotNet.DeviceBinding.CSharp",
+  "name": ".NET Core IoT Device Binding Project",
+  "shortName": "device-binding",
+  "tags": {
+    "language": "C#",
+    "type": "project"
+  },
+  "sourceName": "_DeviceBinding",
```

Comment:
I don't have a lot of experience with dotnet templates, do you know if this supports something like `dotnet new console -n MyCoolProject` which would generate the project name correctly, and will do that for the class as well.

Reply:
This is actually my first template I've developed, as well.  Yes, you run the following after you install the template:

```console
dotnet new device-binding -o DirectoryPath -n NameOfProject
actual name of binding.
```
Where DirectoryPath is the location you want to create the project and NameOfProeject would be the device binding (e.g. Mcp23xxx, etc.)

From what I understand, you can omit -o if you are creating the project in directory you are working.  In most cases, you be pointed to the iot/src/devices directory.

The -n parameter will change all the text within the directory structure to the NameOfProject... including project name, namespaces, classes, etc.

You can see example of a new binding I did using the template [here](https://github.com/shaggygi/iot/tree/Ssd1306-device-binding). 

I just use the follwoing:

```console
dotnet new device-binding -o C:\[path]\iot\src\bindings -n Ssd1306
```